### DISCUSSION TITLE
"SSL_ERROR_INTERNAL_ERROR_ALERT" Encountered during fresh install

### DISCUSSION THREAD

**Discussion by 970261** [nextcloud/all-in-one#5492] [Questions] [answered]
### Steps to reproduce
1.Installed latest Docker 4.34.3 from 4.26 for Windows
2.Followed "Reset the instance" in case old installation still persisted from years ago
3.Ran installation from command after reading and searching through documentation(s)
4.Chose to install Hub 9 instead of 8

### Expected behavior 
Should be able to access Nextcloud via AIO to finish setup

### Actual behavior 
SSL_ERROR_INTERNAL_ERROR_ALERT

#### Host OS
Windows 10 (via Docker WSL)

#### Output of `docker logs nextcloud-aio-apache`
```
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Waiting for Nextcloud to start...
Connection to nextcloud-aio-nextcloud (172.18.0.7) 9000 port [tcp/*] succeeded!
[Mon Oct 28 14:02:40.182319 2024] [mpm_event:notice] [pid 74:tid 74] AH00489: Apache/2.4.62 (Unix) configured -- resuming normal operations
[Mon Oct 28 14:02:40.182465 2024] [core:notice] [pid 74:tid 74] AH00094: Command line: '/usr/local/apache2/bin/httpd -D FOREGROUND'
{"level":"info","ts":1730138560.1970997,"msg":"using config from file","file":"/tmp/Caddyfile"}
{"level":"info","ts":1730138560.198459,"msg":"adapted config to JSON","adapter":"caddyfile"}
{"level":"info","ts":1730138560.2010815,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details."}
{"level":"error","ts":1730138604.644809,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"[domain]","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:dns","title":"","detail":"DNS problem: query timed out looking up CAA for [domain]","instance":"","subproblems":[]}}
{"level":"error","ts":1730138604.6448429,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"[domain]","problem":{"type":"urn:ietf:params:acme:error:dns","title":"","detail":"DNS problem: query timed out looking up CAA for [domain]","instance":"","subproblems":[]},"order":"https://acme-v02.api.letsencrypt.org/acme/order/2025426187/317882577637","attempt":1,"max_attempts":3}
{"level":"error","ts":1730138604.6448572,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"[domain]","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 400 urn:ietf:params:acme:error:dns - DNS problem: query timed out looking up CAA for [domain]"}
{"level":"error","ts":1730138604.6448963,"logger":"tls.obtain","msg":"will retry","error":"[[domain]] Obtain: [w[domain]] solving challenge: [domain]: [w[domain]] authorization failed: HTTP 400 urn:ietf:params:acme:error:dns - DNS problem: query timed out looking up CAA for [domain] (ca=https://acme-v02.api.letsencrypt.org/directory)","attempt":1,"retrying_in":60,"elapsed":44.443299608,"max_duration":2592000}
{"level":"error","ts":1730138695.313366,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"[domain]","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:dns","title":"","detail":"DNS problem: SERVFAIL looking up A for [domain] - the domain's nameservers may be malfunctioning; DNS problem: query timed out looking up AAAA for [domain]","instance":"","subproblems":[]}}
{"level":"error","ts":1730138695.313418,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"w[domain]","problem":{"type":"urn:ietf:params:acme:error:dns","title":"","detail":"DNS problem: SERVFAIL looking up A for [domain] - the domain's nameservers may be malfunctioning; DNS problem: query timed out looking up AAAA for [domain]","instance":"","subproblems":[]},"order":"https://acme-staging-v02.api.letsencrypt.org/acme/order/169009303/20051444753","attempt":1,"max_attempts":3}
{"level":"error","ts":1730138695.3134367,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"[domain]","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 400 urn:ietf:params:acme:error:dns - DNS problem: SERVFAIL looking up A for [domain] - the domain's nameservers may be malfunctioning; DNS problem: query timed out looking up AAAA for [domain]"}
```
#### Docker run command or docker-compose file that you used
```
docker run ^
--init ^
--sig-proxy=false ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
--publish 80:80 ^
--publish 8080:8080 ^
--publish 8443:8443 ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
-e NEXTCLOUD_DATADIR="/run/desktop/mnt/host/e/dockernc" ^
--env NEXTCLOUD_MEMORY_LIMIT=8192M ^
--add-host [domain]:[host IP] ^
nextcloud/all-in-one:latest
```
#### Other valuable info 
Not behind VPN, public IP is available and unchanged, domain was changed from one I had setup AIO on years ago (and removed at the time) to get a feel for WSL.
All ports confirmed open on Host PC via https://canyouseeme.org/

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/discussions/2105

**Comment by davidhund** [comment]
Forgive me if I am hijacking this issue or in the wrong place but… I have followed the default AIO installation instructions from the readme and cannot access Nextcloud.

#2105 and other issues do not shed much light...

I run a local Ubuntu 24 Server with a fixed IP and have added a new subdomain `cloud.mydomain.nl` (`mydomain` is an example😉) at my registar that correctly points to my local router IP. Ports `80 TCP`, `8443 TCP`, `443  TCP/UDP` and `3478 TCP/UDP` are publicly accessible and forwarded from my router to the local server IP. 

The docker Apache logs show noting strange but the `nextcloud-aio-nextcloud` log ends with: 

```
cURL error 28: Operation timed out after 5002 milliseconds with 7986 out of 30787 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://cloud.mydomain.nl/hosting/discovery
```

- `https://<local server IP>:8080/containers` seems fine and shows all containers as running
- `https://<local server IP>` returns a `SSL_ERROR_INTERNAL_ERROR_ALERT` in my browser
- `curl -I http://cloud.mydomain.nl` (`:80`) returns (I assume properly) a 301 by `Caddy` (not Apache?)
- https://cloud.mydomain.nl times out

I have no clue where to look, certs? Firewalls?

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one/discussions/2105

**Reply by davidhund** [reply]
#2105 is not helpful: there are no errors in my log, I am not behind a proxy or CludFlare, I am not in CGNAT and there are no AAAA records.

UPDATE: by chance I noticed the domain _does_ work when accessed from outside my personal network (on mobile over 4G)!?

**Reply by szaimen** [reply]
> by chance I noticed the domain _does_ work when accessed from outside my personal network (on mobile over 4G)!?

I see. Then your router does not support hairpin nat. You can fix this so: https://github.com/nextcloud/all-in-one#how-can-i-access-nextcloud-locally

**Reply by davidhund** [reply]
Thank you for your quick response! I will try and setup pi-hole together with NC.
If I read correctly, this can also be done by adding another community container to the stack: which means adding the community container and re-running the docker command, I guess?
